packages/builder/src/builder.js

```diff
@@ -207,6 +207,7 @@ export default class Builder {
   async generateRoutesAndFiles() {
     consola.debug(`Generating nuxt files`)
 
+    this.plugins.length = 0
     this.plugins.push.apply(this.plugins, this.normalizePlugins())
```

Comment:
Can't we still simply set a new array to this.plugins?

Reply:
This wouldn't clear all references of `this.plugins`

Reply:
`BuildContext` is using it for registering alias in webpack builder.

Reply:
I have an idea, will open a pr.